1.train_models/linear_reg_plate_1_2_norm_data/nbconverted/train_model.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Fit a linear model on cell morphology features
+#
+# Our aim is to determine which features significantly contribute to NF1 genotype adjusted by covariates. This notebook is adapted from Dr. Way at:
+# https://github.com/WayScience/Benchmarking_NF1_data/blob/main/5_analyze_data/notebooks/linear_model/fit_linear_model.ipynb
+
+# In[ ]:
+
+
+from pathlib import Path
+import pandas as pd
+
+from sklearn.linear_model import LinearRegression
+import numpy as np
+from scipy import stats
+
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# In[ ]:
+
+
+sys.path.append(f"{root_dir}/utils")
+import preprocess_utils as ppu
+
+
+# In[ ]:
+
+
+data_path = Path("data")
+
+data_path.mkdir(
+    parents=True, exist_ok=True
+)  # Create the parent directories if they don't exist
+
+
+# In[ ]:
+
+
+plate_path = Path(
+    f"{root_dir}/nf1_painting_repo/3.processing_features/data/normalized_data"
+)
+
+filename1 = "Plate_1_sc_norm.parquet"
+filename2 = "Plate_2_sc_norm.parquet"
+
+path1 = plate_path / filename1
+path2 = plate_path / filename2
+
+po1 = ppu.Preprocess_data(path=path1)
+po2 = ppu.Preprocess_data(path=path2)
+
+plate1df = po1.df  # Returns the dataframe generated by the csv
+plate2df = po2.df  # Returns the dataframe generated by the csv
+
+# Set plate column:
+plate1df["Metadata_plate"] = "1"
+plate2df["Metadata_plate"] = "2"
+
+common_columns = list(plate1df.columns.intersection(plate2df.columns))
+plate1df = plate1df.loc[:, common_columns]
+plate2df = plate2df.loc[:, common_columns]
+
+# Combine the plate dataframes:
+cp_df = pd.concat([plate1df, plate2df], axis="rows")
+cp_df = cp_df.loc[:, (cp_df != 0).any(axis=0)]
+cp_df.reset_index(inplace=True, drop=True)
```

Comment:
You might also consider [pandas chaining](https://towardsdatascience.com/using-pandas-method-chaining-to-improve-code-readability-d8517c5626ac) here. I love chaining in pandas!

```suggestion
cp_df = pd.concat([plate1df, plate2df], axis="rows")
cp_df = (
    cp_df
    .loc[:, (cp_df != 0).any(axis=0)]
    .reset_index(drop=True)
)
```

Reply:
oh wait, it's probably good to not `commit` this suggestion since it's in the nbconverted file.

If you decide to make this change, it belongs in the `.ipynb`

Reply:
Right now I use a specially formatted python file as a jupyter notebook file, so the jupyter notebook is just for reviewers right now.